Code/Framework/AzCore/Tests/Settings/ConfigurableStackTests.cpp

```diff
@@ -0,0 +1,283 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/JSON/document.h>
+#include <AzCore/Memory/SystemAllocator.h>
+#include <AzCore/Settings/ConfigurableStack.h>
+#include <AzCore/Serialization/Json/JsonSerialization.h>
+#include <AzCore/Serialization/Json/RegistrationContext.h>
+#include <AzCore/Serialization/Json/JsonSystemComponent.h>
+#include <AzCore/UnitTest/TestTypes.h>
+
+namespace UnitTest
+{
+    struct ConfigInt
+    {
+        AZ_TYPE_INFO(UnitTest::ConfigInt, "{1FAF6E55-7FA4-4FFA-8C41-34F422B8E8AB}");
+        AZ_CLASS_ALLOCATOR(ConfigInt, AZ::SystemAllocator, 0);
+
+        int m_value;
+
+        static void Reflect(AZ::ReflectContext* context)
+        {
+            if (auto sc = azrtti_cast<AZ::SerializeContext*>(context))
+            {
+                sc->Class<ConfigInt>()->Field("Value", &ConfigInt::m_value);
+            }
+        }
+    };
+
+    struct ConfigurableStackTests : public AllocatorsFixture
+    {
+        void Reflect(AZ::ReflectContext* context)
+        {
+            if (auto sc = azrtti_cast<AZ::SerializeContext*>(context))
+            {
+                AZ::JsonSystemComponent::Reflect(sc);
+                ConfigInt::Reflect(sc);
+                sc->RegisterGenericType<AZ::ConfigurableStack<ConfigInt>>();  
+            }
+            else if (auto jrc = azrtti_cast<AZ::JsonRegistrationContext*>(context))
+            {
+                AZ::JsonSystemComponent::Reflect(jrc);
+            }
+        }
+
+        void SetUp() override
+        {
+            AllocatorsFixture::SetUp();
+
+            m_serializeContext = aznew AZ::SerializeContext();
+            m_jsonRegistrationContext = aznew AZ::JsonRegistrationContext();
+
+            Reflect(m_serializeContext);
+            Reflect(m_jsonRegistrationContext);
+            
+            m_deserializationSettings.m_registrationContext = m_jsonRegistrationContext;
+            m_deserializationSettings.m_serializeContext = m_serializeContext;
+        }
+
+        void TearDown()
+        {
+            m_jsonRegistrationContext->EnableRemoveReflection();
+            Reflect(m_jsonRegistrationContext);
+            m_jsonRegistrationContext->DisableRemoveReflection();
+
+            m_serializeContext->EnableRemoveReflection();
+            Reflect(m_serializeContext);
+            m_serializeContext->DisableRemoveReflection();
+
+            delete m_jsonRegistrationContext;
+            delete m_serializeContext;
```

Comment:
These can be put into a unique_ptr if  this fixture inherits from `ScopedAllocatorFixture`